commit c0901ec6afa75f2aa818de99f06ba64cb2a72bc2 (refs/changes/21/4621/5)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-08-08T18:53:14+02:00 (1 year, 2 months ago)
    
    TOOLS-1962 'sdcadm create sdc -s NEW-HEADNODE' should work
    Reviewed by: Marsell Kukuljevic <marsell@joyent.com>
    Approved by: Marsell Kukuljevic <marsell@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index d97a16b..89e175a 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -12,7 +12,9 @@
 
 ### 1.20.2 (not yet released)
 
+- TOOLS-1962 `sdcadm create sdc -s NEW-HEADNODE` should work
 - TOOLS-1966 `sdcadm create -s dhcpd` should set `dhcpd_server` flag to admin Nic
+- TRITON-636 trouble with timeouts in `sdcadm health`
 
 ## 1.20.1
 
diff --git a/lib/procedures/create-service-instance-v1.js b/lib/procedures/create-service-instance-v1.js
index aaa5333..087d3ff 100644
--- a/lib/procedures/create-service-instance-v1.js
+++ b/lib/procedures/create-service-instance-v1.js
@@ -240,6 +240,28 @@ CreateServiceInstanceV1.prototype.execute = function csiv1Execute(opts, cb) {
             }, next);
         });
 
+        if (change.service.name === 'sdc') {
+            funcs.push(function sdcSvcWarning(_, next) {
+                progress(
+                    '\n' +
+                    'Remember that only a single \'sdc\' VM should run the ' +
+                    'hermes and napi-ufds-watcher services.\nThose will not ' +
+                    'be automatically disabled by sdcadm, nor will the zone ' +
+                    'tools be symlinked.\n' +
+                    'This can be done by running (from the Compute Node ' +
+                    'where the sdc zone has been created):\n\n' +
+                    '    ln -s /zones/${new_uuid}/root/opt/smartdc/sdc ' +
+                    '/opt/smartdc/sdc\n' +
+                    '\n' +
+                    'hermes and napi-ufds-watcher services can be disabled ' +
+                    'with the following commands:\n\n' +
+                    '    svcadm -z ${new_uuid} disable napi-ufds-watcher\n' +
+                    '    svcadm -z ${new_uuid} disable *hermes*\n'
+                );
+                next();
+            });
+        }
+
         opts.log.info({change: change},
                 'CreateServiceInstanceV1 createSvcInst');
         vasync.pipeline({funcs: funcs, arg: arg}, nextSvc);
diff --git a/lib/procedures/index.js b/lib/procedures/index.js
index 948416d..dfc3bf3 100644
--- a/lib/procedures/index.js
+++ b/lib/procedures/index.js
@@ -639,7 +639,7 @@ function coordinatePlan(opts, cb) {
         function createServiceInstance(_, next) {
             // Any instance which should never be created using
             // this tool should be here:
-            var avoid = ['binder', 'manatee', 'rabbitmq', 'sdc', 'zookeeper'];
+            var avoid = ['binder', 'manatee', 'rabbitmq', 'zookeeper'];
 
             var handle = [];
             var remaining = [];
